protocol/get_events.go

```diff
@@ -267,8 +267,8 @@ func (t TopicFilter) Matches(event []xdr.ScVal) bool {
 }
 
 type SegmentFilter struct {
-	Wildcard *string
-	ScVal    *xdr.ScVal
+	Wildcard *string    `json:"-"`
+	ScVal    *xdr.ScVal `json:"-"`
```

Comment:
I believe adding the "-" is unnecessary. Doing so tells the default marshal implementation to skip marshalling the field. However, the type has a custom marshal function, so the default impl does not read these tags. Is that correct or do they still get used somehow?

Ref: https://pkg.go.dev/encoding/json#Marshal

Reply:
Yeah, I think you’re right. I added these while first developing prior to the interface implementation just to hide ‘em. I personally think it’s helpful as a reader to see that and think “okay lemme find how this is actually serialized,” but also happy to remove it due to its redundancy.